File: RetailOrdering/RetailOrdering.Application/DTOs/ProductDto.cs
namespace RetailOrdering.Application.DTOs
{
    public class ProductDto
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal BasePrice { get; set; }
        public string Category { get; set; } = string.Empty;
        public string Brand { get; set; } = string.Empty;
        public string ImageUrl { get; set; } = string.Empty;
        public int AvailableStock { get; set; }
        public int LowStockThreshold { get; set; }
        public string SizeOptionsJson { get; set; } = "[]";
        public string ToppingOptionsJson { get; set; } = "[]";
        public bool IsAvailable { get; set; }
        public bool IsStockLow { get; set; }
    }
}

File: RetailOrdering/retail-ordering-ui/src/app/core/services/product-dto.ts
export interface SizeOption {
  Name: string;
  PriceModifier: number;
}

export interface ToppingOption {
  Name: string;
  Price: number;
}

export interface ProductDto {
  id: number;
  name: string;
  description: string;
  basePrice: number;
  category: string;
  brand: string;
  imageUrl: string;
  availableStock: number;
  lowStockThreshold: number;
  sizeOptionsJson: string;
  toppingOptionsJson: string;
  isAvailable: boolean;
  isStockLow: boolean;

  // Optional parsed types for frontend convenience
  sizeOptions?: SizeOption[];
  toppingOptions?: ToppingOption[];
}
